README.md

```diff
@@ -18,90 +18,122 @@ Multiplex requires a JSON file with the following format:
         {
             "name": "common-params",
             "params":
+                [
+                    {
+                        "arg": "bs", "vals": [ "4k", "8k" ], "role": "client"
+                    },
+                    {
+                        "arg": "rw", "vals": [ "read", "write" ]
+                    }
+                ]
+        }
+    ],
+    "sets":
+        [
             [
                 {
-                    "arg": "bs", "vals": [ "4k", "8k" ]
+                    "include": "common-params"
                 },
                 {
-                    "arg": "rw", "vals": [ "read", "write" ]
+                    "arg": "ioengine", "vals": [ "sync" ]
                 }
             ]
-        }
-    ],
-    "sets":
-    [
-    [
-        {
-            "include": "common-params"
-        },
-        {
-            "arg": "ioengine", "vals": [ "sync" ]
-        }
-    ]
-    ]
+        ]
 }
 ```
 
-The `global-options` and the `sets` sections are both required.
-A set of "multi-value parameters" is included in each set.  Each set, combined
-with the multi-value paramters in "global-options", will be used to expand to a new set
-of single-value parameters to STDOUT, like the sample generated in
-`JSON/bench-params-output.json`:
+### global-options
+The `global-options` section is required. It contains blocks of general configuration
+data can be replicated and included in the `sets` section.
+In this section, you may have an array of multi-value parameters that are common to
+the test run as "common-params". Multi-value params are specified with the `args` and
+`vals` keywords. The `role` key is optional and defaults to 'client' if omitted. Valid
+roles are: "client", "server" and "all".
+
+Also, you may have a block to define benchmark or tooling specific settings. Example:
+"crucible-defaults".
```

Comment:
I think this does not go here.  This idea is for a multiplex.json (or some other name) that would be part of the workload itself -- not something that a user provides.

@atheurer agree?

Reply:
> I think this does not go here. This idea is for a multiplex.json (or some other name) that would be part of the workload itself -- not something that a user provides.
> 
> @atheurer agree?

I took crucible-defaults block from fio json example. Did I misunderstood its purpose ?

Reply:
So I think the "defaults" are something that the workload itself defines (in a multiplex.json that probably doesn't currently exist).